tests/Behat/WebtoolsCleanupContext.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_webtools\Behat;
+
+use Behat\Behat\Hook\Scope\AfterScenarioScope;
+use Behat\Behat\Hook\Scope\BeforeScenarioScope;
+use Behat\Gherkin\Node\ScenarioInterface;
+use Drupal\Core\Cache\Cache;
+use Drupal\Core\Path\AliasStorage;
+use Drupal\DrupalExtension\Context\RawDrupalContext;
+
+/**
+ * Context to clean up entities created through the UI during scenarios.
+ */
+class WebtoolsCleanupContext extends RawDrupalContext {
+
+  /**
+   * The IDs of the existing entities in the system, keyed by entity type.
+   *
+   * @var array
+   */
+  protected $existing = [];
+
+  /**
+   * The PIDs of the existing url aliases.
+   *
+   * @var array
+   */
+  protected $existingAliases = [];
+
+  /**
+   * Collect the IDs of entities existing before the execution of the scenario.
+   *
+   * Entity types can be marked for cleanup by adding a tag that starts with
+   * "cleanup:" followed by the entity type machine name.
+   *
+   * @param \Behat\Behat\Hook\Scope\BeforeScenarioScope $scope
+   *   The scenario scope.
+   *
+   * @beforeScenario
+   */
+  public function collectExistingEntities(BeforeScenarioScope $scope): void {
+    // Reset the entity list at the beginning of each scenario.
+    $this->existing = [];
+
+    $entity_types = $this->getEntityTypesToCleanup($scope->getScenario());
+    foreach ($entity_types as $entity_type) {
+      $this->existing[$entity_type] = $this->getAllEntityIdsOfType($entity_type);
+    }
+  }
+
+  /**
+   * Collect the PIDs of aliases existing before the execution of the scenario.
+   *
+   * @param \Behat\Behat\Hook\Scope\BeforeScenarioScope $scope
+   *   The scenario scope.
+   *
+   * @beforeScenario @cleanup-aliases
+   */
+  public function collectExistingUrlAliases(BeforeScenarioScope $scope): void {
+    // Reset the alias list at the beginning of each scenario.
+    $this->existingAliases = [];
+    // Executing database query,
+    // because AliasStorage->load() returns only a single url alias.
```

Comment:
Comment can go in one line.

Reply:
I did rephrase. Initially I used second line for avoiding code sniffer error.